src/screens/GroupInfo/elements/ModalAction.js

```diff
@@ -38,12 +38,12 @@ const ModalAction = (props) => {
         return (
           <>
             <TouchableOpacity
-              onPress={() => onPress('message-anonymously', selectedUser)}
+              onPress={() => onPress('message', selectedUser)}
               style={styles.buttonStyle}>
-              <Text
-                style={selectedUser?.allow_anon_dm ? styles.textButton : styles.disabledTextButton}>
-                Message Incognito
-              </Text>
+              <View style={{flexDirection: 'row'}}>
+                <Text style={styles.textButton}>Leave Incognito Mode</Text>
+                <Loading visible={isLoadingInitChat} />
+              </View>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a loading indicator `isLoadingInitChat` which is a good practice for better user experience. However, it's not clear from the provided context if `isLoadingInitChat` is properly managed (set to true when chat initiation starts and set to false when it ends). Please ensure that this state variable is being correctly updated in the relevant parts of your code.

Also, the button text has been changed from "Message Incognito" to "Leave Incognito Mode". This seems to suggest a change in functionality, but the onPress function still triggers a 'message' action. If the intention is to leave the incognito mode, the function associated with the onPress event should reflect that.

```diff
-  onPress={() => onPress('message', selectedUser)}
+  onPress={() => onPress('leaveIncognitoMode', selectedUser)}
```

Please make sure that the 'leaveIncognitoMode' action is properly handled in the onPress function.


<!-- This is an auto-generated comment by OSS CodeRabbit -->